pkg/sync/sync_context.go

```diff
@@ -153,7 +153,7 @@ func WithResourceModificationChecker(enabled bool, diffResults *diff.DiffResultL
 	}
 }
 
-// WithNamespaceCreation will create non-exist namespace
+// WithNamespaceCreation will create a namespace, if it does not already exist
```

Comment:
What would happen if the namespace already exists? How will it handle cases when labels/annotations exist in live state with different values than provided in desired state? In this case, it should update the labels/annotations values.

Reply:
I clarified the semantics a bit in my latest commit. Do you think that this makes sense (both in wording and semantics)? 😄